json-unit-core/src/main/java/net/javacrumbs/jsonunit/core/internal/Node.java

```diff
@@ -98,6 +98,10 @@ public String getDescription() {
 
     BigDecimal decimalValue();
 
+    default boolean isInteger() {
+        return decimalValue().scale() <= 0;
```

Comment:
Not sure whether scale is enough to ensure whether a given value is an integer https://stackoverflow.com/questions/1078953/check-if-bigdecimal-is-integer-value.